bakta_queries_unittest.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import sqlite3
+import os
+from tempfile import TemporaryDirectory
+from bakta_queries import (
+    connect_to_database,
+    close_database_connection,
+    get_gene_info_by_name,
+    extract_gene_sequence_from_fasta,
+    write_gene_sequences_to_fasta,
+    extract_gene_sequence_to_fasta,
+)
+
+class TestYourScript(unittest.TestCase):
+    def setUp(self):
+        # Create a temporary directory for test output files
+        self.temp_dir = TemporaryDirectory()
+        self.db_connection = sqlite3.connect(':memory:')  # In-memory database
+        self.db_cursor = self.db_connection.cursor()
+
+        # Create the necessary tables for testing
+        self.create_test_tables()
+
+    def tearDown(self):
+        # Clean up temporary files and directories
+        self.temp_dir.cleanup()
+        # Close the in-memory database connection
+        self.db_connection.close()
+
+    def test_connect_to_database(self):
+        conn, cursor = connect_to_database(self.db_connection)
+        self.assertIsInstance(conn, sqlite3.Connection)
+        self.assertIsInstance(cursor, sqlite3.Cursor)
+
+    def test_close_database_connection(self):
+        conn, _ = connect_to_database(self.db_connection)  # Fixed this line
+        close_database_connection(conn)
+        with self.assertRaises(sqlite3.ProgrammingError):
+            conn.execute("SELECT 1")
+
+    def create_test_tables(self):
+        # Create the 'bakta' and 'identifier' tables for testing
+        self.db_cursor.execute(
+            """
+            CREATE TABLE IF NOT EXISTS bakta (
+                contig_id TEXT,
+                start INT,
+                end INT,
+                strand TEXT,
+                entity_id TEXT,
+                gene_id TEXT,
+                filename TEXT,
+                filepath TEXT
+            )
+            """
+        )
+
+        self.db_cursor.execute(
+            """
+            CREATE TABLE IF NOT EXISTS identifier (
+                entity_id TEXT,
+                filename TEXT,
+                filepath TEXT
+            )
+            """
+        )
+
```

Comment:
The schema should be put into an external .sql file. But, let's consider that out of scope for right now, as we have to get machinery inplace to handle the .sql files. One of the challenges with databases is they evolve overtime. For example, look at how many patches we have with [microsetta](https://github.com/biocore/microsetta-private-api/tree/master/microsetta_private_api/db/patches)